packages/dds/sequence/src/sharedStringWithSequentialId.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IMergeTreeOp, Marker,
+    MergeTreeDeltaType, reservedMarkerIdKey } from "@fluidframework/merge-tree";
+import { ISequencedDocumentMessage, MessageType } from "@fluidframework/protocol-definitions";
+import { parseHandles } from "@fluidframework/shared-object-base";
+import { createIdAfterMin, createIdBeforeMax } from "./generateSequentialId";
+import { SharedString } from "./sharedString";
+
+export function sharedStringWithSequentialIdMixin(Base: typeof SharedString = SharedString): typeof SharedString {
+    return class SequenceWithSequentialId extends Base {
+        protected processMergeTreeMsg(
+            rawMessage: ISequencedDocumentMessage): void {
+                const message = parseHandles(
+                    rawMessage,
+                    this.runtime.IFluidSerializer);
+                this.applySequentialId(message);
+                super.processMergeTreeMsg(rawMessage);
+        }
+
+        private applySequentialId(msg: ISequencedDocumentMessage) {
+            if (msg.type !== MessageType.Operation) {
+                return;
+            }
+
+            const op = msg.contents as IMergeTreeOp;
+            if (op.type === MergeTreeDeltaType.INSERT && op.seg !== undefined
```

Comment:
When can we have MergeTreeDeltaType.INSERT but either not have segment or have it not object? If never, then this should be assert (which is runtime check that throws in runtime) vs. IF condition.